[Writing Tools] Weak link Writing Tools instead of soft linking

rr-codes opened the pull request:
#### 118689a19bbdf460450dbae60bbe1da687f0f04f
<pre>
[Writing Tools] Weak link Writing Tools instead of soft linking
<a href="https://bugs.webkit.org/show_bug.cgi?id=275541">https://bugs.webkit.org/show_bug.cgi?id=275541</a>
<a href="https://rdar.apple.com/129933357">rdar://129933357</a>

Reviewed by Megan Gardner.

* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h: Added.
* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsUISPI.h: Added.
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:

Canonical link: <a href="https://commits.webkit.org/280059@main">https://commits.webkit.org/280059@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1491e9002f232fe5f5119dd1c2f2e90e51cbd716

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55645 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34968 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8112 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58629 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6075 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57771 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42590 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6274 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/4175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57674 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32870 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/47962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/25943 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29656 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/5291 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4219 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51603 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/5558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60220 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30799 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/5685 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/52241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31884 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/48032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51729 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32965 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8205 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31631 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
```diff
@@ -3,12 +3,14 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 55;
```
fix this

megangardner replied:
Wait, you're taking this out right?

megangardner reviewed (approved):
Just wait for my patch to land (which is literally in flight) and rebase the OTHER_LDFLAGS, etc.

rr-codes committed:
[Writing Tools] Weak link Writing Tools instead of soft linking
https://bugs.webkit.org/show_bug.cgi?id=275541
rdar://129933357

Reviewed by Megan Gardner.

* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h: Added.
* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsUISPI.h: Added.
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:

Canonical link: https://commits.webkit.org/280059@main

emw-apple commented on Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h:
```diff
@@ -25,6 +25,17 @@
+
+#else
+
+#error Symbols must be forward declared once used with non-internal SDKS.
```
This is kind of a weird admission to check in to the open-source project—let's make sure the work to declare these SPI symbols is tracked.
